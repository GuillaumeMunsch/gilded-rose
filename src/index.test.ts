import { GildedRose } from ".";
import runGoldenMaster from "jest-golden-master";

describe("Non regression tests", () => {
  it("Should test the backstage pass several times", () => {
    runGoldenMaster(async () => {
      // GIVEN
      const gildedRose = new GildedRose([
        { name: "Backstage passes to a TAFKAL80ETC concert", quality: 25, sellIn: 15 },
      ]);

      // WHEN
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
    });
  });

  it("Should test sulfuras several times", () => {
    runGoldenMaster(async () => {
      // GIVEN
      const gildedRose = new GildedRose([{ name: "Sulfuras, Hand of Ragnaros", quality: 80, sellIn: 15 }]);

      // WHEN
      gildedRose.updateQuality();
    });
  });

  it("Should test aged brie several times", () => {
    runGoldenMaster(async () => {
      // GIVEN
      const gildedRose = new GildedRose([{ name: "Aged Brie", quality: 25, sellIn: 15 }]);

      // WHEN
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
    });
  });

  it("Should test default items several times", () => {
    runGoldenMaster(async () => {
      // GIVEN
      const gildedRose = new GildedRose([{ name: "Default item", quality: 25, sellIn: 15 }]);

      // WHEN
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
      gildedRose.updateQuality();
    });
  });
});
