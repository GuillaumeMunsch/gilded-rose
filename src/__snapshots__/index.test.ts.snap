// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Non regression tests Should test aged brie several times 1`] = `
[
  "Items on construction: [
Name: Aged Brie
Quality: 25
]",
  "Items pre updateQuality: [
Name: Aged Brie
Quality: 25
]",
  "End of updateQuality: [
Name: Aged Brie
Quality: 26
]",
  "Items pre updateQuality: [
Name: Aged Brie
Quality: 26
]",
  "End of updateQuality: [
Name: Aged Brie
Quality: 27
]",
  "Items pre updateQuality: [
Name: Aged Brie
Quality: 27
]",
  "End of updateQuality: [
Name: Aged Brie
Quality: 28
]",
  "Items pre updateQuality: [
Name: Aged Brie
Quality: 28
]",
  "End of updateQuality: [
Name: Aged Brie
Quality: 29
]",
  "Items pre updateQuality: [
Name: Aged Brie
Quality: 29
]",
  "End of updateQuality: [
Name: Aged Brie
Quality: 30
]",
  "Items pre updateQuality: [
Name: Aged Brie
Quality: 30
]",
  "End of updateQuality: [
Name: Aged Brie
Quality: 31
]",
  "Items pre updateQuality: [
Name: Aged Brie
Quality: 31
]",
  "End of updateQuality: [
Name: Aged Brie
Quality: 32
]",
  "Items pre updateQuality: [
Name: Aged Brie
Quality: 32
]",
  "End of updateQuality: [
Name: Aged Brie
Quality: 33
]",
  "Items pre updateQuality: [
Name: Aged Brie
Quality: 33
]",
  "End of updateQuality: [
Name: Aged Brie
Quality: 34
]",
  "Items pre updateQuality: [
Name: Aged Brie
Quality: 34
]",
  "End of updateQuality: [
Name: Aged Brie
Quality: 35
]",
  "Items pre updateQuality: [
Name: Aged Brie
Quality: 35
]",
  "End of updateQuality: [
Name: Aged Brie
Quality: 36
]",
  "Items pre updateQuality: [
Name: Aged Brie
Quality: 36
]",
  "End of updateQuality: [
Name: Aged Brie
Quality: 37
]",
  "Items pre updateQuality: [
Name: Aged Brie
Quality: 37
]",
  "End of updateQuality: [
Name: Aged Brie
Quality: 38
]",
  "Items pre updateQuality: [
Name: Aged Brie
Quality: 38
]",
  "End of updateQuality: [
Name: Aged Brie
Quality: 39
]",
  "Items pre updateQuality: [
Name: Aged Brie
Quality: 39
]",
  "End of updateQuality: [
Name: Aged Brie
Quality: 40
]",
  "Items pre updateQuality: [
Name: Aged Brie
Quality: 40
]",
  "End of updateQuality: [
Name: Aged Brie
Quality: 42
]",
  "Items pre updateQuality: [
Name: Aged Brie
Quality: 42
]",
  "End of updateQuality: [
Name: Aged Brie
Quality: 44
]",
  "Items pre updateQuality: [
Name: Aged Brie
Quality: 44
]",
  "End of updateQuality: [
Name: Aged Brie
Quality: 46
]",
  "Items pre updateQuality: [
Name: Aged Brie
Quality: 46
]",
  "End of updateQuality: [
Name: Aged Brie
Quality: 48
]",
  "Items pre updateQuality: [
Name: Aged Brie
Quality: 48
]",
  "End of updateQuality: [
Name: Aged Brie
Quality: 50
]",
  "Items pre updateQuality: [
Name: Aged Brie
Quality: 50
]",
  "End of updateQuality: [
Name: Aged Brie
Quality: 50
]",
  "Items pre updateQuality: [
Name: Aged Brie
Quality: 50
]",
  "End of updateQuality: [
Name: Aged Brie
Quality: 50
]",
  "Items pre updateQuality: [
Name: Aged Brie
Quality: 50
]",
  "End of updateQuality: [
Name: Aged Brie
Quality: 50
]",
  "Items pre updateQuality: [
Name: Aged Brie
Quality: 50
]",
  "End of updateQuality: [
Name: Aged Brie
Quality: 50
]",
  "Items pre updateQuality: [
Name: Aged Brie
Quality: 50
]",
  "End of updateQuality: [
Name: Aged Brie
Quality: 50
]",
  "Items pre updateQuality: [
Name: Aged Brie
Quality: 50
]",
  "End of updateQuality: [
Name: Aged Brie
Quality: 50
]",
  "Items pre updateQuality: [
Name: Aged Brie
Quality: 50
]",
  "End of updateQuality: [
Name: Aged Brie
Quality: 50
]",
  "Items pre updateQuality: [
Name: Aged Brie
Quality: 50
]",
  "End of updateQuality: [
Name: Aged Brie
Quality: 50
]",
  "Items pre updateQuality: [
Name: Aged Brie
Quality: 50
]",
  "End of updateQuality: [
Name: Aged Brie
Quality: 50
]",
  "Items pre updateQuality: [
Name: Aged Brie
Quality: 50
]",
  "End of updateQuality: [
Name: Aged Brie
Quality: 50
]",
  "Items pre updateQuality: [
Name: Aged Brie
Quality: 50
]",
  "End of updateQuality: [
Name: Aged Brie
Quality: 50
]",
]
`;

exports[`Non regression tests Should test default items several times 1`] = `
[
  "Items on construction: [
Name: Default item
Quality: 25
]",
  "Items pre updateQuality: [
Name: Default item
Quality: 25
]",
  "End of updateQuality: [
Name: Default item
Quality: 24
]",
  "Items pre updateQuality: [
Name: Default item
Quality: 24
]",
  "End of updateQuality: [
Name: Default item
Quality: 23
]",
  "Items pre updateQuality: [
Name: Default item
Quality: 23
]",
  "End of updateQuality: [
Name: Default item
Quality: 22
]",
  "Items pre updateQuality: [
Name: Default item
Quality: 22
]",
  "End of updateQuality: [
Name: Default item
Quality: 21
]",
  "Items pre updateQuality: [
Name: Default item
Quality: 21
]",
  "End of updateQuality: [
Name: Default item
Quality: 20
]",
  "Items pre updateQuality: [
Name: Default item
Quality: 20
]",
  "End of updateQuality: [
Name: Default item
Quality: 19
]",
  "Items pre updateQuality: [
Name: Default item
Quality: 19
]",
  "End of updateQuality: [
Name: Default item
Quality: 18
]",
  "Items pre updateQuality: [
Name: Default item
Quality: 18
]",
  "End of updateQuality: [
Name: Default item
Quality: 17
]",
  "Items pre updateQuality: [
Name: Default item
Quality: 17
]",
  "End of updateQuality: [
Name: Default item
Quality: 16
]",
  "Items pre updateQuality: [
Name: Default item
Quality: 16
]",
  "End of updateQuality: [
Name: Default item
Quality: 15
]",
  "Items pre updateQuality: [
Name: Default item
Quality: 15
]",
  "End of updateQuality: [
Name: Default item
Quality: 14
]",
  "Items pre updateQuality: [
Name: Default item
Quality: 14
]",
  "End of updateQuality: [
Name: Default item
Quality: 13
]",
  "Items pre updateQuality: [
Name: Default item
Quality: 13
]",
  "End of updateQuality: [
Name: Default item
Quality: 12
]",
  "Items pre updateQuality: [
Name: Default item
Quality: 12
]",
  "End of updateQuality: [
Name: Default item
Quality: 11
]",
  "Items pre updateQuality: [
Name: Default item
Quality: 11
]",
  "End of updateQuality: [
Name: Default item
Quality: 10
]",
  "Items pre updateQuality: [
Name: Default item
Quality: 10
]",
  "End of updateQuality: [
Name: Default item
Quality: 8
]",
  "Items pre updateQuality: [
Name: Default item
Quality: 8
]",
  "End of updateQuality: [
Name: Default item
Quality: 6
]",
  "Items pre updateQuality: [
Name: Default item
Quality: 6
]",
  "End of updateQuality: [
Name: Default item
Quality: 4
]",
  "Items pre updateQuality: [
Name: Default item
Quality: 4
]",
  "End of updateQuality: [
Name: Default item
Quality: 2
]",
  "Items pre updateQuality: [
Name: Default item
Quality: 2
]",
  "End of updateQuality: [
Name: Default item
Quality: 0
]",
  "Items pre updateQuality: [
Name: Default item
Quality: 0
]",
  "End of updateQuality: [
Name: Default item
Quality: 0
]",
  "Items pre updateQuality: [
Name: Default item
Quality: 0
]",
  "End of updateQuality: [
Name: Default item
Quality: 0
]",
  "Items pre updateQuality: [
Name: Default item
Quality: 0
]",
  "End of updateQuality: [
Name: Default item
Quality: 0
]",
  "Items pre updateQuality: [
Name: Default item
Quality: 0
]",
  "End of updateQuality: [
Name: Default item
Quality: 0
]",
  "Items pre updateQuality: [
Name: Default item
Quality: 0
]",
  "End of updateQuality: [
Name: Default item
Quality: 0
]",
  "Items pre updateQuality: [
Name: Default item
Quality: 0
]",
  "End of updateQuality: [
Name: Default item
Quality: 0
]",
  "Items pre updateQuality: [
Name: Default item
Quality: 0
]",
  "End of updateQuality: [
Name: Default item
Quality: 0
]",
  "Items pre updateQuality: [
Name: Default item
Quality: 0
]",
  "End of updateQuality: [
Name: Default item
Quality: 0
]",
  "Items pre updateQuality: [
Name: Default item
Quality: 0
]",
  "End of updateQuality: [
Name: Default item
Quality: 0
]",
  "Items pre updateQuality: [
Name: Default item
Quality: 0
]",
  "End of updateQuality: [
Name: Default item
Quality: 0
]",
  "Items pre updateQuality: [
Name: Default item
Quality: 0
]",
  "End of updateQuality: [
Name: Default item
Quality: 0
]",
]
`;

exports[`Non regression tests Should test sulfuras several times 1`] = `
[
  "Items on construction: [
Name: Sulfuras, Hand of Ragnaros
Quality: 45
]",
  "Items pre updateQuality: [
Name: Sulfuras, Hand of Ragnaros
Quality: 45
]",
  "End of updateQuality: [
Name: Sulfuras, Hand of Ragnaros
Quality: 45
]",
]
`;

exports[`Non regression tests Should test the backstage pass several times 1`] = `
[
  "Items on construction: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 25
]",
  "Items pre updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 25
]",
  "End of updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 26
]",
  "Items pre updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 26
]",
  "End of updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 27
]",
  "Items pre updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 27
]",
  "End of updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 28
]",
  "Items pre updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 28
]",
  "End of updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 29
]",
  "Items pre updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 29
]",
  "End of updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 30
]",
  "Items pre updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 30
]",
  "End of updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 32
]",
  "Items pre updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 32
]",
  "End of updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 34
]",
  "Items pre updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 34
]",
  "End of updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 36
]",
  "Items pre updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 36
]",
  "End of updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 38
]",
  "Items pre updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 38
]",
  "End of updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 40
]",
  "Items pre updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 40
]",
  "End of updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 43
]",
  "Items pre updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 43
]",
  "End of updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 46
]",
  "Items pre updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 46
]",
  "End of updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 49
]",
  "Items pre updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 49
]",
  "End of updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 50
]",
  "Items pre updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 50
]",
  "End of updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 50
]",
  "Items pre updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 50
]",
  "End of updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 0
]",
  "Items pre updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 0
]",
  "End of updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 0
]",
  "Items pre updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 0
]",
  "End of updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 0
]",
  "Items pre updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 0
]",
  "End of updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 0
]",
  "Items pre updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 0
]",
  "End of updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 0
]",
  "Items pre updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 0
]",
  "End of updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 0
]",
  "Items pre updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 0
]",
  "End of updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 0
]",
  "Items pre updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 0
]",
  "End of updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 0
]",
  "Items pre updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 0
]",
  "End of updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 0
]",
  "Items pre updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 0
]",
  "End of updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 0
]",
  "Items pre updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 0
]",
  "End of updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 0
]",
  "Items pre updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 0
]",
  "End of updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 0
]",
  "Items pre updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 0
]",
  "End of updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 0
]",
  "Items pre updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 0
]",
  "End of updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 0
]",
  "Items pre updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 0
]",
  "End of updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 0
]",
  "Items pre updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 0
]",
  "End of updateQuality: [
Name: Backstage passes to a TAFKAL80ETC concert
Quality: 0
]",
]
`;
